activerecord/test/cases/transaction_callbacks_test.rb

```diff
@@ -367,6 +367,14 @@ def test_after_commit_callbacks_should_validate_on_condition
     assert_match(/:on conditions for after_commit and after_rollback callbacks have to be one of \[:create, :destroy, :update\]/, e.message)
   end
 
+  def test_after_commit_callbacks_can_be_disabled_when_restoring_state
+    record = TopicWithCallbacks.create!
+    callback_called = false
+    record.after_commit_block { callback_called = true }
+    record.committed!(should_run_callbacks: false)
```

Comment:
Since `commited!` is an internal API, it is preferable to test the fix by this PR from public API.